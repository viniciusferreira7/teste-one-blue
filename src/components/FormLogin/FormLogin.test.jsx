import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { FormLogin } from '.';

describe('<FormLogin/>', () => {
  it('should render FormLogin', async () => {
    await act(async () => render(<FormLogin />));
    const formLogin = screen.getByRole('heading', { name: 'Entrar' });
    expect(formLogin).toBeInTheDocument();
  });

  it('should render inputs and button', async () => {
    await act(async () => render(<FormLogin />));
    expect(
      screen.getByPlaceholderText('Digite seu nome de usuário'),
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Digite sua senha')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Entrar' })).toBeInTheDocument();
  });

  it('should match snapshot', async () => {
    await act(async () => render(<FormLogin />));
    expect(
      screen.getByPlaceholderText('Digite seu nome de usuário').parentElement,
    ).toMatchSnapshot();
  });
});
