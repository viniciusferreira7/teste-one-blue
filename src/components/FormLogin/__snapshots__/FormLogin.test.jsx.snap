// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLogin/> should match snapshot 1`] = `
<form
  class="sc-bczRLJ bFCfUJ"
>
  <h1>
    Entrar
  </h1>
  <p
    class="sc-jSMfEi jDxcCH"
  >
    Necessário preencher o campo de Senha
  </p>
  <input
    class="sc-gsnTZi hukAyA"
    name="name"
    placeholder="Digite seu nome de usuário"
    required=""
    type="text"
  />
  <input
    class="sc-dkzDqf iXafZv"
    name="password"
    placeholder="Digite sua senha"
    required=""
    type="password"
  />
  <button
    class="sc-hKMtZM gnuIkq"
  >
    Entrar
  </button>
</form>
`;
